util/chplenv/defaults.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python
+"""
+Check .chplconfig file and environment variables for default value overrides
+"""
+from __future__ import print_function
+import os
+import sys
+from sys import stderr
+
+chplenv_dir = os.path.dirname(__file__)
+sys.path.insert(0, os.path.abspath(chplenv_dir))
+
+# Global dictionary that will be populated w/ .chplconfig file contents
+chplconfig = {}
+
+# Globally track if we've parsed already
+parsed = False
+
+# Valid options for override by .chplconfig file
+CHPLS = [
+         'CHPL_HOME',
+         'CHPL_HOST_PLATFORM',
+         'CHPL_HOST_COMPILER',
+         'CHPL_TARGET_PLATFORM',
+         'CHPL_TARGET_COMPILER',
+         'CHPL_TARGET_ARCH',
+         'CHPL_LOCALE_MODEL',
+         'CHPL_COMM',
+         'CHPL_COMM_SUBSTRATE',
+         'CHPL_GASNET_SEGMENT',
+         'CHPL_TASKS',
+         'CHPL_LAUNCHER',
+         'CHPL_TIMERS',
+         'CHPL_UNWIND',
+         'CHPL_MEM',
+         'CHPL_MAKE',
+         'CHPL_ATOMICS',
+         'CHPL_NETWORK_ATOMICS',
+         'CHPL_GMP',
+         'CHPL_HWLOC',
+         'CHPL_REGEXP',
+         'CHPL_WIDE_POINTERS',
+         'CHPL_LLVM',
+         'CHPL_AUX_FILESYS',
+         ]
+
+
+def parse():
+    """ Parse .chplconfig file for acceptable env var overrides """
+
+    # Default initialized to empty dictionary
+    chplconfig = {}
+
+    # Potential paths of .chplconfig file
+    chplconfigpath = os.environ.get('CHPL_CONFIG')
```

Comment:
Added a `_find()` and `_prettypath()` (since that was defined when searching for path)